kong/plugins/ai-proxy/drivers/openai.lua

```diff
@@ -0,0 +1,179 @@
+local _M = {}
+
+-- imports
+local cjson = require("cjson.safe")
+local fmt = string.format
+local ai_shared = require("kong.plugins.ai-proxy.drivers.shared")
+local socket_url = require "socket.url"
+--
+
+local transformers_to = {
+  ["llm/v1/chat"] = function(request_table, model, max_tokens, temperature, top_p)
+    -- if user passed a prompt as a chat, transform it to a chat message
+    if request_table.prompt then
+      request_table.messages = {
+        {
+          role = "user",
+          content = request_table.prompt,
+        }
+      }
+    end
+  
+    local this = {
+      model = model,
+      messages = request_table.messages or nil,
+      max_tokens = max_tokens or nil,
+      temperature = temperature or nil,
+      top_p = top_p or nil,
+    }
+  
+    return this, "application/json", nil
+  end,
+
+  ["llm/v1/completions"] = function(request_table, model, max_tokens, temperature, top_p)
+    local this = {
+      prompt = request_table.prompt or nil,
+      model = model,
+      max_tokens = max_tokens or nil,
+      temperature = temperature or nil,
+    }
```

Comment:
Just an FYI: the `or nil` is not strictly necessary:

```suggestion
    local this = {
      prompt = request_table.prompt,
      model = model,
      max_tokens = max_tokens,
      temperature = temperature,
    }
```
(the only difference is that if one of these values is `false` then you'll get a `false` value instead of `nil`, which may be your motivation)

The extra verbosity here doesn't bother me in the slightest, so no need to change these things unless you would like to.

Reply:
I corrected this everywhere where nil checks won't fall through./